drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -1554,6 +1564,39 @@ if hue.datastore["dni_to_device_id"] == nil then
   hue.datastore["dni_to_device_id"] = {}
 end
 
+
+if hue.datastore["api_keys"] == nil then
+  hue.datastore["api_keys"] = {}
+end
+
+if Discovery.api_keys and next(Discovery.api_keys) ~= nil then
+  local datastore_dirty = false
+  for k, v in pairs(Discovery.api_keys) do
+    hue.datastore.api_keys[k] = v
+    datastore_dirty = true
+  end
+
+  if datastore_dirty then
+    hue.datastore:save()
+  end
+end
```

Comment:
What is the point of the `datastore_dirty` flag? It doesn't seem to be doing anything here. A follow up is that I dont fully understand what is trying to be accomplished by this section that runs prior to the driver being run, and prior to the metatable being set on the Discovery.api_keys table. Is it the goal to copy any previously Discovered api keys into the datastore? Will `Discovery.api_keys` ever have an entry when this code is run?

Reply:
It's overly-defensive coding after all the odd things I've seen happen.

Reply:
> Will Discovery.api_keys ever have an entry when this code is run?

Prior to the run loop of the driver, this is impossible, so I would suggest removing it all tbh.

Reply:
I was thinking about instances in the past where the Discovery loop could be loaded multiple times simultaneously, but thinking about it harder, the package path cache would prevent that from being problematic here anyway. I've removed it.